### ISSUE TITLE
AGG palette reading regressed main menu

### ISSUE THREAD

**Issue by jdm** [ihhub/fheroes2#4639]
After 9029fd6c, the main menu no longer appears after I skip the opening video. I just get a black screen with the menu music playing.

**Comment by oleg-derevenetz** [comment]
Hi @jdm what is your configuration (I mean, OS and SDL version)?

**Comment by jdm** [comment]
I'm on macOS, using SDL 2. When I build with DEBUG=1 and run under lldb and set a breakpoint during startup, the problem didn't appear. I'm going to keep poking to figure out what's going on.

**Comment by jdm** [comment]
Hmm, simply running a DEBUG=1 build doesn't show the issue, whether a debugger is involved or not.

**Comment by jdm** [comment]
And then the problem disappeared after another rebuild. That's very worrying :/

**Comment by oleg-derevenetz** [comment]
It looks like #4638 is related to that change too.

**Comment by oleg-derevenetz** [comment]
No, I found the cause for #4638 and it seems that it's not related to this issue. BTW, I was not able to reproduce this issue so far on my macOS.

**Comment by ihhub** [comment]
AGG palette reading shouldn't affect anything. It might be that your build was cached and wasn't built properly. Can you check again that exact commit with clean build?

**Comment by ihhub** [comment]
If intro is played correctly then palette is set in a right way and it doesn't change for main menu.

**Comment by jdm** [comment]
The symptoms I had were:
* I pulled the latest master revision and performed `make clean`, followed by `make`
* the intro video played, then the main menu and cursor did not appear but the music played
* when I pressed escape and then enter (to confirm the invisible dialog), the program quit appropriately

When I reverted 9029fd6c7e6148a8488a63834e68d7960bea30ab and performed `make clean`, followed by `make`, the resulting build played the intro video and then showed me the menu.

When I re-applied 9029fd6c7e6148a8488a63834e68d7960bea30ab and performed `make clean`, followed by `make`, the menu was invisible once again. Then I ran `make clean` and `make DEBUG=1`, and the build worked fine. When I did another `make clean` and `make`, that build also ran fine. I have not been able to reproduce the problem since then.

I'm fine with closing this issue at this point until someone experiences the issue once more.
